cluster/src/main/java/io/scalecube/cluster/gossip/GossipProtocol.java

```diff
@@ -48,35 +48,45 @@
 public final class GossipProtocol implements IGossipProtocol, IManagedGossipProtocol {
   private static final Logger LOGGER = LoggerFactory.getLogger(GossipProtocol.class);
 
-  private ITransport transport;
+  private final String memberId;
+  private final ITransport transport;
+  private final ScheduledExecutorService executor;
+  private final GossipProtocolSettings settings;
+
   @SuppressWarnings("unchecked")
   private Subject<Message, Message> subject = new SerializedSubject(PublishSubject.create());
   private Map<String, GossipLocalState> gossipsMap = Maps.newHashMap();
-  private volatile List<TransportEndpoint> members = new ArrayList<>();
-  private final TransportEndpoint localEndpoint;
-  private final ScheduledExecutorService executor;
+  private volatile List<Address> members = new ArrayList<>();
+
   private AtomicLong counter = new AtomicLong(0);
   private Queue<GossipTask> gossipsQueue = new ConcurrentLinkedQueue<>();
-  private int maxGossipSent = 2;
-  private int gossipTime = 300;
-  private int maxEndpointsToSelect = 3;
   private long period = 0;
   private volatile int factor = 1;
   private Subscriber<Message> onGossipRequestSubscriber;
   private ScheduledFuture<?> executorTask;
 
-  public GossipProtocol(TransportEndpoint localEndpoint) {
-    this.localEndpoint = localEndpoint;
-    this.executor = createDedicatedScheduledExecutor();
+  /**
+   * Creates new instance of gossip protocol with given memberId, transport and default settings.
+   *
+   * @param memberId id of current member
+   * @param transport transport
+   */
+  public GossipProtocol(String memberId, ITransport transport) {
+    this(memberId, transport, GossipProtocolSettings.DEFAULT);
   }
 
-  public GossipProtocol(TransportEndpoint localEndpoint, ScheduledExecutorService executor) {
-    this.localEndpoint = localEndpoint;
-    this.executor = executor;
-  }
-
-  private ScheduledExecutorService createDedicatedScheduledExecutor() {
-    return Executors.newScheduledThreadPool(1, createThreadFactory("scalecube-gossip-scheduled-%s"));
+  /**
+   * Creates new instance of gossip protocol with given memberId, transport and settings.
+   *
+   * @param memberId id of current member
+   * @param transport transport
+   * @param settings gossip protocol settings
+   */
+  public GossipProtocol(String memberId, ITransport transport, GossipProtocolSettings settings) {
+    this.memberId = memberId;
+    this.transport = transport;
+    this.settings = settings;
+    this.executor = Executors.newSingleThreadScheduledExecutor(createThreadFactory("scalecube-gossip-scheduled-%s"));
```

Comment:
In analogy with ``BootstrapFactory`` this ``"scalecube-gossip-scheduled-%s"`` name must be ``"sc-gossip-%s"``